ui-tests/src/test/java/io/jenkins/plugins/coverage/UiTest.java

```diff
@@ -0,0 +1,256 @@
+package io.jenkins.plugins.coverage;
+
+import org.jenkinsci.test.acceptance.junit.AbstractJUnitTest;
+import org.jenkinsci.test.acceptance.plugins.git.GitScm;
+import org.jenkinsci.test.acceptance.po.Build;
+import org.jenkinsci.test.acceptance.po.FreeStyleJob;
+import org.jenkinsci.test.acceptance.po.Job;
+
+import io.jenkins.plugins.coverage.CoveragePublisher.Adapter;
+import io.jenkins.plugins.coverage.CoveragePublisher.CoveragePublisher;
+import io.jenkins.plugins.coverage.CoveragePublisher.CoveragePublisher.SourceFileResolver;
+import io.jenkins.plugins.coverage.CoveragePublisher.Threshold.AdapterThreshold.AdapterThresholdTarget;
+import io.jenkins.plugins.coverage.CoveragePublisher.Threshold.GlobalThreshold.GlobalThresholdTarget;
+
+/**
+ * Base class for all UI tests. Provides several helper methods that can be used by all tests.
+ */
+class UiTest extends AbstractJUnitTest {
+    static final String JACOCO_ANALYSIS_MODEL_XML = "jacoco-analysis-model.xml";
+    static final String JACOCO_CODINGSTYLE_XML = "jacoco-codingstyle.xml";
+    static final String JACOCO_FROM_COMMIT_XML = "jacoco-f52a5169.xml";
+    static final String RESOURCES_FOLDER = "/";
```

Comment:
I would recommend to create a specific folder in the same package, otherwise all test class files are copied as well:

```
Building in workspace /Users/hafner/Development/git/warnings-ng-plugin-devenv/code-coverage-api-plugin/ui-tests/target/jenkins12905174398993849233home/workspace/thirsty_summary
[thirsty_summary] $ /bin/sh -xe /var/folders/pg/qr8ry2kd4qjc151jhq9ksgx00000gn/T/jenkins14812671326808423143.sh
+ base64 --decode
+ unzip -o archive.zip
Archive:  archive.zip
  inflating: jacoco-f52a5169.xml     
  inflating: jacoco-analysis-model.xml  
   creating: io/
   creating: io/jenkins/
   creating: io/jenkins/plugins/
   creating: io/jenkins/plugins/coverage/
  inflating: io/jenkins/plugins/coverage/CoverageSummaryTest.class  
  inflating: io/jenkins/plugins/coverage/UiTest$InCaseCoverageDecreasedConfiguration.class  
  inflating: io/jenkins/plugins/coverage/CoverageReportTest.class  
  inflating: io/jenkins/plugins/coverage/SmokeTests.class  
  inflating: io/jenkins/plugins/coverage/UiTest.class  
   creating: io/jenkins/plugins/coverage/util/
  inflating: io/jenkins/plugins/coverage/util/TrendChartTestUtil.class  
  inflating: io/jenkins/plugins/coverage/util/ChartUtilAssert.class  
  inflating: io/jenkins/plugins/coverage/CoverageReportTabAssert.class  
  inflating: io/jenkins/plugins/coverage/MainPanelTest.class  
  inflating: io/jenkins/plugins/coverage/UiTest$JobConfiguration.class  
  inflating: io/jenkins/plugins/coverage/CoverageReportAssert.class  
  inflating: io/jenkins/plugins/coverage/Assertions.class  
  inflating: io/jenkins/plugins/coverage/MainPanelAssert.class  
  inflating: io/jenkins/plugins/coverage/FileCoverageTableHeaderAssert.class  
  inflating: io/jenkins/plugins/coverage/SoftAssertions.class  
  inflating: io/jenkins/plugins/coverage/SourceCodeViewAssert.class  
   creating: io/jenkins/plugins/coverage/CoveragePublisher/
  inflating: io/jenkins/plugins/coverage/CoveragePublisher/CoveragePublisherCoveragePublisherSourceFileResolverAssert.class  
  inflating: io/jenkins/plugins/coverage/CoveragePublisher/CoveragePublisherCoveragePublisherAssert.class  
   creating: io/jenkins/plugins/coverage/CoveragePublisher/Threshold/
  inflating: io/jenkins/plugins/coverage/CoveragePublisher/Threshold/CoveragePublisherThresholdGlobalThresholdGlobalThresholdTargetAssert.class  
  inflating: io/jenkins/plugins/coverage/CoveragePublisher/Threshold/CoveragePublisherThresholdAdapterThresholdAdapterThresholdTargetAssert.class  
  inflating: io/jenkins/plugins/coverage/CoveragePublisher/Threshold/CoveragePublisherThresholdAdapterThresholdAssert.class  
  inflating: io/jenkins/plugins/coverage/CoveragePublisher/Threshold/CoveragePublisherThresholdGlobalThresholdAssert.class  
  inflating: io/jenkins/plugins/coverage/CoveragePublisher/CoveragePublisherAdapterAssert.class  
  inflating: io/jenkins/plugins/coverage/CoveragePublisherTest.class  
  inflating: io/jenkins/plugins/coverage/CoverageSummaryAssert.class  
  inflating: io/jenkins/plugins/coverage/FileCoverageTableRowAssert.class  
  inflating: io/jenkins/plugins/coverage/UiTest$ThresholdLevel.class  
  inflating: io/jenkins/plugins/coverage/UiTest$InCaseNoReportsConfiguration.class  
  inflating: jacoco-codingstyle.xml  
Publishing Coverage report....
```